lib/id3c/db/__init__.py

```diff
@@ -214,15 +214,34 @@ def upsert_sample(db: DatabaseSession,
         "collection_identifier": collection_identifier,
         "collection_date": collection_date,
         "encounter_id": encounter_id,
-        "additional_details": Json(additional_details),
+        "additional_details": Json(additional_details) if additional_details else None,
     }
 
     # Look for existing sample(s)
     with db.cursor() as cursor:
         cursor.execute("""
-            select sample_id as id, identifier, collection_identifier, encounter_id
-              from warehouse.sample
-             where identifier = %(identifier)s
+            select
+                sample_id as id, identifier, collection_identifier, encounter_id, details,
+                row (
+                    identifier,
+                    collection_identifier
+                )::text !=
+                row (
+                    %(identifier)s,
+                    %(collection_identifier)s
+                )::text as identifiers_changed,
+                row(
+                    collected::timestamp,
+                    encounter_id,
+                    details
+                )::text !=
+                row(
+                    %(collection_date)s::timestamp,
+                    %(encounter_id)s::integer,
```

Comment:
Still seeing records with `metadata_changed = True`, resulting in an update, when data is not changing. These two lines should also coalesce with the existing values, to match how the record would appear after the update if it occurs.